shared/components/error-boundary/ErrorBoundary.js

```diff
@@ -0,0 +1,35 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+export default class ErrorBoundary extends Component {
+
+  static propTypes = {
+    children: PropTypes.node,
+  }
+
+  state = {
+    error: null,
+    errorInfo: null,
+  };
+
+  componentDidCatch(error, errorInfo) {
+    this.setState({
+      error,
+      errorInfo,
+    });
+  }
+
+  render() {
+    if (this.state.errorInfo) {
+      return (
+        <div>
+          <h2>Error</h2>
+          <p>{this.state.error && this.state.error.toString()}</p>
+          <pre>{this.state.errorInfo.componentStack}</pre>
```

Comment:
What is `componentStack`?

Reply:
It's contains the stack of components that caused the throw, see https://reactjs.org/docs/error-boundaries.html#componentdidcatch-parameters